README.md
# DS-LoadMusicStick
Synology DS App um Musikdateien auf einen USB Stick für einen CD Wechsel zu laden

Mehr Details: https://www.synology-forum.de/threads/neue-app-loadmusicstick.58610/

Vorherige Lösung: https://github.com/Starwhooper/RandomFileCopy

Version 3.3 build 201604##
* fix issue with 0byte files
* apptype change to popup
* end of support for DSM 4

Version 3.2 build 20160421
* send mail after job
* consider musicfiles outside the typical "music" folder

Version 3.1 build 20160418
* fix filesize interpretation since DSM 6
* fix layout of header
* link for feedback via e-mail or forum added
* align foldersize and filecount to right
* extent destination storagesize number 
* fix issue to recognize running job under DSM 6
* disable function to delete files from destination.
* create folder by genre

Version 3.0 build 20151003
* use MediaServer Database instead to seek file by self
* seperate deletion to own skript
* offer Genres as selection
* check if USB is really mount (in the past was only check if the mointpoint folder exist)
* change select all/none checkboxes to java function
* filetype will self detected and need no predefination
* supports now also eSATA as destination
* CD01-CD06 option provides only 50 files per folder (more could the radio not handle)
* convert mp3 to lower rates
* list dates and convert info in logfile
* remember selected sources from previous session
* remember selected configuration from previous session

Version 2.2 build 20141117
* implement delete function
* Add full licence text during installation

Version 2.1
* link to filemanager

Version 2.0
* create bash script to handle to long working task of copy and convert and monitor it

Version 1.11
* remember selection of folder from /volume1/music
* implement button to check/uncheck all folder in /volume1/music
* implement option, to break at first time when a file is to big
* implement option to set a limit of free space
* fix issue with chars like ÄÜÖ on the destination file
* implement option to delete the files from the destination (but it is not active)

Version 1.10
* show in german, if user set languagesetting to german

Version 1.9
* add more errormessage to provide more detaisl in case of errors

Version 1.8 build
* make it compatible with DSM 4.0-2265 and newer version
* add this CHANGELOG document
* note in case of no write permission on USB drive

Version 1.7 build 20141018
* first public release


-------------------requirements------------------------------------
* Init3rdParty (php 5.3 or newer) https://www.cphub.net/ 
* MediaServer (Synology package)

-------------------todo for next version------------------------------------

*texte im protocol übersetzen
* multiuser, damit auch nicht Admin account die App nutzen kann
* Im protokoll die „war zu groß“ meldung weniger gefährlich anzeigen und mehr im design das „Added to job“ darstellen
* reload button um seite zu aktualisieren

== install/uninstall process ==
* implement a log to check over the package manager when the app does 
* implement a upgrade function
* implement to offer a new version if avaiable
* remove checkbox if you want start the app after installation on DMS 4

== GUI ==
* prompt anything during the copy process, maybe show the id3 image during the copy
* translate all textes
* refresh size on destination USB after delete the content

== Icons ==
** implement icon for DSM 4
** implement icon for DSM 4 package managaer
** implement big icon for package manager

== filename output ==
* check chars in destination filename, implement option to remove all non standard chars

== copy options ==
copy in folder differd by genre
copy only 50 files, if it choose (currently it copy more)



More new ideas: May you post here http://www.synology-forum.de/showthread.html?58610-neue-App-quot-LoadMusicStick-quot or send it my via mail to thiemo@schuff.eu


code/CHANGELOG
Version 3.3 build 201604##
* fix issue with 0byte files
* apptype change to popup
* end of support for DSM 4

Version 3.2 build 20160421
* send mail after job
* consider musicfiles outside the typical "music" folder

Version 3.1 build 20160418
* fix filesize interpretation since DSM 6
* fix layout of header
* link for feedback via e-mail or forum added
* align foldersize and filecount to right
* extent destination storagesize number 
* fix issue to recognize running job under DSM 6
* disable function to delete files from destination.
* create folder by genre

Version 3.0 build 20151003
* use MediaServer Database instead to seek file by self
* seperate deletion to own skript
* offer Genres as selection
* check if USB is really mount (in the past was only check if the mointpoint folder exist)
* change select all/none checkboxes to java function
* filetype will self detected and need no predefination
* supports now also eSATA as destination
* CD01-CD06 option provides only 50 files per folder (more could the radio not handle)
* convert mp3 to lower rates
* list dates and convert info in logfile
* remember selected sources from previous session
* remember selected configuration from previous session

Version 2.2 build 20141117
* implement delete function
* Add full licence text during installation

Version 2.1
* link to filemanager

Version 2.0
* create bash script to handle to long working task of copy and convert and monitor it

Version 1.11
* remember selection of folder from /volume1/music
* implement button to check/uncheck all folder in /volume1/music
* implement option, to break at first time when a file is to big
* implement option to set a limit of free space
* fix issue with chars like ÄÜÖ on the destination file
* implement option to delete the files from the destination (but it is not active)

Version 1.10
* show in german, if user set languagesetting to german

Version 1.9
* add more errormessage to provide more detaisl in case of errors

Version 1.8 build
* make it compatible with DSM 4.0-2265 and newer version
* add this CHANGELOG document
* note in case of no write permission on USB drive

Version 1.7 build 20141018
* first public release


-------------------requirements------------------------------------
* Init3rdParty (php 5.3 or newer) https://www.cphub.net/ 
* MediaServer (Synology package)

-------------------todo for next version------------------------------------

*texte im protocol übersetzen
* multiuser, damit auch nicht Admin account die App nutzen kann
* Im protokoll die „war zu groß“ meldung weniger gefährlich anzeigen und mehr im design das „Added to job“ darstellen
* reload button um seite zu aktualisieren

== install/uninstall process ==
* implement a log to check over the package manager when the app does 
* implement a upgrade function
* implement to offer a new version if avaiable
* remove checkbox if you want start the app after installation on DMS 4

== GUI ==
* prompt anything during the copy process, maybe show the id3 image during the copy
* translate all textes
* refresh size on destination USB after delete the content

== Icons ==
** implement icon for DSM 4
** implement icon for DSM 4 package managaer
** implement big icon for package manager

== filename output ==
* check chars in destination filename, implement option to remove all non standard chars

== copy options ==
copy in folder differd by genre
copy only 50 files, if it choose (currently it copy more)



More new ideas: May you post here http://www.synology-forum.de/showthread.html?58610-neue-App-quot-LoadMusicStick-quot or send it my via mail to thiemo@schuff.eu

code/package/index.php
<?php
//////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////
////        YOU NEED TO INSTALL "Init 3rdparty" on your DS.       ////
////         http://www.cphub.net/index.php?id=40&pid=293         ////
//////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////

error_reporting(E_ALL);

include('functions.inc.php');
include('language.cfg.php');
include('config.inc.php');

//phpinfo();


////OUTPUT HEADER
echo '<html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">';
echo '<link rel="stylesheet" href="format.css" type="text/css">';

echo '	<script type="text/javascript" language="JavaScript">
			function togglefolders(source) {
				checkboxes = document.getElementsByName(\'folders[]\');
				for(var i=0, n=checkboxes.length;i<n;i++) {
					checkboxes[i].checked = source.checked;
				}
			}			
			function togglegenres(source) {
				checkboxes = document.getElementsByName(\'genres[]\');
				for(var i=0, n=checkboxes.length;i<n;i++) {
					checkboxes[i].checked = source.checked;
				}
			}
		</script>';

echo '</head><body>';

////////////////////////////////////////////////////////////////////////IN CASE OF NO OPTION (DEFAULT PAGE)
if (!isset($para['option'])){
	echo '<h1>'.ttt('Copy Soundfiles to USB Stick').'</h1>';
	
//////Show details, ins case that copyprogress on work
	if ($sysdetails['dsmversion']['majorversion'] == 6) exec('ps ax|grep /tmp/lms.copy',$cmddump);
	else exec('ps|grep /tmp/lms.copy',$cmddump);
	foreach($cmddump as $dump){
		if((strpos($dump,'sh /tmp/lms.copy') != false) and (strpos($dump,'&&') == false)) {
			echo '<p style="color: red;">'.ttt('actual in progress') . ': '.$dump;
			$scriptname = substr($dump,strrpos($dump,' '));
//			echo ' <a href="log/'.basename($scriptname).'.log.html" target="_blank">log</a>';
	echo ' &lt;-- '.ttt('copy in progress').'</p>';
		}
	}

	mkdir('log');
	
	echo '<form name="input" action="prepare2copy.php?" method="get">';

////POSIBLE FOLDER
	if ($sysdetails['dsmversion']['majorversion'] == 6) $sizemultiplier = 1000000000;
	else $sizemultiplier = 1000000;
	
	$folderlist = dbquery("select round(sum(filesize) / ".$sizemultiplier." ,2) as size, count(id) as count, split_part(path,'/',2) as volume, split_part(path,'/',3) as folder,split_part(path,'/',4) as subfolder, concat('/',split_part(path,'/',2),'/',split_part(path,'/',3),'/',split_part(path,'/',4)) as folderpath from music group by volume, folder, subfolder order by folderpath");
	
	foreach($folderlist as $key => $folder){
//		$folderlist[$key]['folderpath'] = '/'.$folder['volume'].'/'.$folder['folder'].'/'.$folder['subfolder'];
		if (in_array($folderlist[$key]['folderpath'],$settings['selectedfolder'])) $folderlist[$key]['checked'] = ' checked';
		else $folderlist[$key]['checked'] = NULL;
	}
	
	echo '<h2>'.ttt('source').'</h2>';
	echo '<table><tr><td>';
	echo '<p>'.ttt('Select the recognize folders from').' <strong>'.$sourceroot.'</strong> '.ttt('that contains the files that you want copy randomized to the USB Stick').'.</p>';
	
	echo '<table><tr><th>'.ttt('folder').'</th><th>'.ttt('size').'</th><th>'.ttt('count').'</th></tr>';
	if(count($folderlist) >= 1) {
		foreach($folderlist as $folder) {
			echo '<tr><td nowrap><input type="checkbox" name="folders[]" value="'.$folder['folderpath'].'"'.$folder['checked'].'><div style="display:inline;font-size:xx-small">/'.$folder['volume'].'/'.$folder['folder'].'/</div>'.$folder['subfolder'].'</td><td class="right">'.$folder['size'].'GB</td><td class="right">'.$folder['count'].'</td></tr>';
		}
		echo '<tr><td nowrap><input type="checkbox" onClick="togglefolders(this)" /><strong>'.ttt('all').'/'.ttt('none').'</strong></td></tr>';	
	}
	else echo msg('','',ttt('no source avaiable, please check if folder').' "'.$sourceroot.'" '.ttt('contains subfolders'),'red');

	echo '</table>';
	echo '</p>';	

////POSSIBLE EXTENSIONS	
	echo '</td><td><p>'.ttt('audioformats').':<p>';
	$formats = dbquery("select lower(right(path,strpos(reverse(path),'.')-1)) as extension, count(id) as count from music group by extension order by extension");
	foreach($formats as $format) {
		echo '<input type="checkbox" name="exts[]" value="'.$format['extension'].'"';
		if ($format['extension'] == 'mp3') echo ' checked';
		echo '>'.$format['extension'].' ('.$format['count'].')<br />';
	}
	echo '</p></td>';
	
//POSSIBLE GENRES
	echo '</td><td><p>'.ttt('genres').':<p>';
	$genres = dbquery("select genre, count(id) as count from music group by genre order by lower(genre)");

	$i = 0;
	foreach($genres as $genre) {
		$genrelists[$i] .= '<input type="checkbox" name="genres[]" value="'.$genre['genre'].'"';
		if (in_array($genre['genre'],$settings['selectedgenre'])) $genrelists[$i] .= ' checked';
		else $genrelists[$i] .= '';
		$genrelists[$i] .= '>'.$genre['genre'].' ('.$genre['count'].')<br />';
		$i++;
		if ($i == 4) $i = 0;
	}
	
	echo '<table><tr>';
	foreach($genrelists as $genrelist) echo '<td nowrap>'.$genrelist.'</td>';
	echo '<tr><td nowrap><input type="checkbox" onClick="togglegenres(this)" /><strong>'.ttt('all').'/'.ttt('none').'</strong></td></tr>';	
	echo '</tr></table>';
	echo '</p></td></tr></table>';
	
	


////LIST AVAIABLE DESTINATIONS
	echo '<h2>'.ttt('destionations').'	</h2><p>'.ttt('Select the destination of your music files').'.</p>';
	
	$drives = checkdestinations();

	$check = ' checked';
	
	if ($drives == NULL) echo msg('','',ttt('NO PORTABLE DRIVE AS DESTINATION FOUND'),'red');	
	else{
		foreach($drives as $folder => $detail){
			echo '<p><input type="radio" name="volume" value="'.$folder.'"'.$check.'>'.ttt('free space on destination').' <strong>'.$folder.':</strong> '.$detail['freegb'].'Gbyte of '.$detail['totalgb'].'Gbyte - '.ttt('That means').' '.$detail['freegrade'] .'% '.ttt('are available');
		//	if($detail['writeable'] == false) echo ' '.ttt('But').' <font color="red">'.ttt('NO WRITE PERMISSION').'</font>';
				
//			echo ' <a href="'.$approot.'/delete.php?drive='.$folder.'" target="Blank">'.ttt('delete content').'</a>'; doesn't work wich too much files
			echo '</p>';
			$check = NULL;
		}
	}
	echo ' <p><a href="'.$dsmurl.'/webman/index.cgi?launchApp=SYNO.SDS.App.FileStation3.Instance" target="Blank">'.ttt('open synology filemanager').'</a></p>';
		
	
	
////SHOW OPTIONS TO MANUPULATE THE COPY	
	if ($drives != NULL){
	echo '<h2>'.ttt('option').'</h2><p>'.ttt('Settings for the kind of copy').'</p>';
	echo '<p><input type="checkbox" name="differbygenre" value="true">'.ttt('differfolderbygenre').' <strong>'.ttt('or').'</strong> <input type="checkbox" name="CD16" value="true">'.ttt('Copy 50 files in each Folder CD01 - CD06 to use it with a CD changer Simulator for older Carradios. As example when you use the Yatour Digital Music Changer.').'</p>';
	echo '<p></p>';
//	echo '<p><input type="checkbox" name="baftbf" value="true">'.ttt('Breakes the copyjob at first to big file').'</p>';
//	echo '<p><input type="checkbox" name="leftfreespace" value="true">'.ttt('Left free space on destination').': <input type="text" name="leftfreespacemb">MB</p>';
	echo '<p><input type="checkbox" name="convert2mp3" value="true">'.ttt('Convert Files to MP3');
	echo ' (change bitrate: <select name="bitrate" size="1">';
	echo '<option>no</option>';
	foreach($allowedbitrate as $bitrate) echo '<option>'.$bitrate.'</option>';
	echo '</select>)';
	echo '</p>';	
	
	
////SHOW THE START BUTTON	
	echo '<h2>'.ttt('lets go').'</h2><p><input type="submit" name="option" target="_blank" value="'.ttt('Start to copy').'"></form></p><p style="font-size:xx-small;">'.ttt('after start').'</p>';
	
	}
}

////FEEDBACK
echo '<h2>Feedback</h2><p>'.ttt('FEEDBACK').'</p>';
	
////IMPRESSUM
echo '<hr><p><table style="border-style:none;"><tr style="border-style:none;"><td style="border-style:none;">'.ttt('Autor').': <a href="http://thiemo.schuff.eu">Thiemo Schuff</a> - April 2016</td><td style="text-align:right;border-style:none;"><a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nd/4.0/88x31.png" /></a><br />'.ttt('This work is licensed under a').' <a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/">Creative Commons Attribution-NoDerivatives 4.0 International '.ttt('License').'</a></td></tr></table></p>.';
	
	
?>

code/package/delete.php
<?php
error_reporting(E_ALL);

include('functions.inc.php');
include('language.cfg.php');
include('config.inc.php');


if(isset($para['drive'])){
	$drive = $para['drive'];
	if(file_exists($drive)){
//		if (in_array($drive,$allowedusbdrives)){
			$cmd = 'find '.$drive;
			exec($cmd,$cmddump);
			echo '<form action="delete.php">';
			foreach($cmddump as $file){
				if(strlen($file) <= strlen($drive) + 1) continue;
				echo '<input type="checkbox" name="files[]" value="'.$file.'"  checked="checked">'.$file.'</br>';
			}
			echo '<input type="submit" value="delete"></form>';
//		}
//		else exit('no allowed drive');
	}
	else exit(ttt('no exist drive'));
}
elseif(isset($para['files'])){
	$files = $para['files'];
//	var_dump($files);
	foreach($files as $file){
		unlink($file);
		echo 'delete: '.$file.'<br/>';
	}
}
else exit('no selected drive');
?>

code/package/prepare2copy.php
<?php
error_reporting(E_ALL);

include('functions.inc.php');
include('language.cfg.php');
include('config.inc.php');

//var_dump($para);

////////////////////////////////////////////////////////////////////////IN CASE OF OPTION "START TO COPY", that will you get in case you clicked on the button on the default page
if($para['option'] == ttt('Start to copy')){
	
	$jobscript = getcwd().'/log/jobscript_'.date('YmdGis').'.txt';

	////CANCEL IN CASE OF NO DESTINATION	
	if ($para['volume'] == NULL) {
		echo msg('*',$para['volume'],ttt('destination is not allowed, maybe you forgot to choose a destination or the Stick is not recognize from DS'),'red');
		continue;
	}
	else $destvolume = $para['volume'].'/';

//	foreach($para['folders'] as $folder) $dest .= $folder.', ';
	echo '<h1>'.ttt('startcopyprocess').'</h1><p><a href="index.php">'.ttt('back to startpage').'</a><small>'.ttt('please check progressinformation on startpage').'</small></p><h2>'.ttt('Protocol').'</h2>';

	if (!isset($para['folders'])){
		echo msg('*',$para['volume'],ttt('destination is not allowed, maybe you forgot to choose a destination or the Stick is not recognize from DS'),'red');
		continue;
	}

	if (!isset($para['exts'])){
		echo msg('*','*',ttt('no file extension choosen'),'red');
		continue;
	}
		
	//SAVE SETTINGS
	savesettings($settingsfile,$para);

	////LIST ALL USABLE FILES FROM SOURCE FOLDERS
	$query = "SELECT right(path,strpos(reverse(path),'/')-1) as filename, path, concat('/',split_part(path,'/',2),'/',split_part(path,'/',3),'/',split_part(path,'/',4)) as folderpath, lower(right(path,strpos(reverse(path),'.')-1)) as extension, filesize, genre FROM music ORDER by random()";
	$files = dbquery($query);

	////LIST ALL USABLE FILES FROM SOURCE FOLDERS
	//CREATE PHP SCRIPT TO LOAD STICK
	if($para['leftfreespace'] == 'true') $copyjobheader['leftfreespace'] = '$para[\'leftfreespacemb\'] = '.$para['leftfreespacemb'].';';


	if ($para['CD16'] == true) {
		for($i=1;$i<=6;$i++) $copyjobscript[] = 'if (is_dir("'.$destvolume.'CD0'.$i.'") == false) mkdir("'.$destvolume.'CD0'.$i.'");';
	}
	$i = 0;

	foreach ($files as $key => $file){
		
		$destionationfile = $destinationfolder = NULL;
		
		if(!in_array($file['folderpath'],$para['folders'])) continue;
		if(!in_array($file['genre'],$para['genres'])) continue;
		if(!in_array($file['extension'],$para['exts'])) continue;
		
		$j=0;
		
		if ($para['CD16'] == true and $j == 50) continue(2);
//		echo $usablespaceofonstick.':'.$file['filesize']."<br>";
		
//		if($usablespaceofonstick) < $file['filesize']) {
//			if ($para['baftbf'] == true) continue(2);
//			continue;
//		}
		
		$destionationfile = $file['filename'];

		$destionationfile = str_pad($key+1, 5, 0, STR_PAD_LEFT).'_'.$destionationfile;

		if ($para['CD16'] == true) {
			$i++;
			$destinationfolder = 'CD0'.$i.'/';
			if ($i == 6) {
				$i = 0;
				$j++;
			}
		}
		elseif ($para['differbygenre'] == true) {
			if ($file['genre'] == '') $file['genre'] = '_none';
			$copyjobscript[] = 'if (is_dir("'.$destvolume.$file['genre'].'") == false) mkdir("'.$destvolume.$file['genre'].'");';
			$destinationfolder = $file['genre'].'/';
		}
		else $destinationfolder = NULL;

		if ($file['extension'] != 'mp3' and $para['convert2mp3'] == true) {
			if(isset($para['bitrate']) and $para['bitrate'] != 'no') $bitrate='-b:a'.$para['bitrate'].'k';
			else $bitrate = NULL;
			$newfilename .= '.mp3';
			//$copyjobscript[] = 'exec("/usr/syno/bin/ffmpeg -n -i '.escapeshellarg($file['path']).' '.$bitrate.' '.escapeshellarg($destvolume.'/'.$destinationfolder.'/'.$destionationfile).'.mp3")';
			$copyjobscript[] = 'exec("/usr/syno/bin/ffmpeg -n -i '.$file['path'].' '.$bitrate.' '.$destvolume.$destinationfolder.$destionationfile.'.mp3");';
//			$script[] = 'echo $(date +%Y-%m-%d\ %H:%M:%S)'.escapeshellarg($destvolume.'/'.$newfilename).' done (convert)'."\n".'>>'.$shellscriptlog;
//			$output .= '<li>'.ttt('Added to job to convert').': '.$file['path'].' to '.$destvolume.'/'.$newfilename.'</li>';
			$j++;
		}
		else {
//			$copyjobscript[] = 'copy '.escapeshellarg($file['path']).' '.escapeshellarg($destvolume.'/'.$newfilename);
			$copyjobscript[] = 'if (disk_free_space("'.$destvolume.'") >= '.$file['filesize'].') copy("'.$file['path'].'","'.$destvolume.$destinationfolder.$destionationfile.'");';
//			$script[] = 'echo $(date +%Y-%m-%d\ %H:%M:%S)'.escapeshellarg($destvolume.'/'.$newfilename).' done'."\n".'>>'.$shellscriptlog;
//			$usablespaceofonstick = $usablespaceofonstick - $file['filesize'];
//			$output .= '<li>'.ttt('Added to job').': '.$file['path'].' to '.$destvolume.'/'.$newfilename.'</li>';
			$j++;
		}
	}

//	$copyjobscript[] = 'rm -f '.$shellscript;


	$handler = fopen($jobscript, 'w');
	foreach($copyjobheader as $line) fwrite($handler , $line."\n");
	foreach($copyjobscript as $line) fwrite($handler , $line."\n");
	fclose($handler);

	//$cmd = 'chmod 777 '.$shellscript.' && sh '.$shellscript.' >/dev/null 2>/dev/null &';
//	$cmd = 'php copy.php '.$jobscript.' > /dev/null &';
	$cmd = 'php copy.php '.$jobscript.' > '.$jobscript.'.log';
	exec($cmd);

	echo ttt('work_in_progress').': '.$jobscript;	
//	echo $output;	
	

}

?>

code/package/language.cfg.php
<?php
$lang['eng'] = array(
	'after start' => 'after clicking on button its not needed to wait till the website response. You may close the browser and wait for confirmation mail.',
	'or' => 'or',
	'differfolderbygenre' => 'differ files in folder by genre',
	'actual in progress' => 'actual in progress',
	'all' => 'all',
	'any issue' => 'any issue',
	'are available' => 'are available',
	'are to big for destination' => 'are to big for destination',
	'audioformats' => 'audioformats',
	'autor' => 'autor',
	'breakes the copyjob at first to big file' => 'breakes the copyjob at first to big file',
	'but only' => 'but only',
	'but' => 'but',
	'clean' => 'clean',
	'contains subfolders' => 'contains subfolders',
	'convert files to mp3' => 'convert files to mp3',
	'copy 50 files in each folder cd01 - cd06 to use it with a cd changer simulator for older carradios. as example when you use the yatour digital music changer.' => 'copy 50 files in folder cd01 - cd06 to use it with a cd changer simulator for older carradios. as example when you use the yatour digital music changer.',
	'copy soundfiles to usb stick' => 'copy soundfiles to usb stick',
	'could not copy' => 'could not copy',
	'count' => 'count',
	'delete content' => 'delete content',
	'destination file are not writable' => 'destination file are not writable',
	'destination folder not exist' => 'destination folder not exist',
	'destination is not allowed, maybe you forgot to choose a destination or the stick is not recognize from ds' => 'destination is not allowed, maybe you forgot to choose a destination or the stick is not recognize from ds',
	'destionations' => 'destionations',
	'done' => 'copy done',
	'feedback' => '<a href="mailto:thiemo@schuff.eu&subject=loadmusicstick&body=Hi Thiemo,...">E-Mail</a> or <a href="http://www.synology-forum.de/showthread.html?58610-neue-App-LoadMusicStick" target="_blank">Forum</a>',
	'file size is' => 'file size is',
	'file' => 'file',
	'folder' => 'folder',
	'free space on destination' => 'free space on destination',
	'genres' => 'genres',
	'in reason of' => 'in reason of',
	'left free space on destination' => 'left free space on destination',
	'lets go' => 'lets go',
	'license' => 'license',
	'no folders with sources files choosen' => 'no folders with sources files choosen',
	'no portable drive as destination found' => 'no portable drive as destination found',
	'no source avaiable, please check if folder' => 'no source avaiable, please check if folder',
	'no write permission' => 'no write permission',
	'none' => 'none',
	'open synology filemanager' => 'open synology filemanager',
	'option' => 'option',
	'permission' => 'permission',
	'protocol' => 'protocol',
	'select the destination of your music files' => 'select the destination of your music files',
	'select the recognize folders from' => 'select the recognize folders from',
	'select' => 'select',
	'settings for the kind of copy' => 'settings for the kind of copy',
	'size' => 'size',
	'source file is not a file' => 'source file is not a file',
	'source file is not readable' => 'source file is not readable',
	'source file now exist' => 'source file now exist',
	'source' => 'source',
	'start to copy' => 'start to copy',
	'that contains the files that you want copy randomized to the usb stick' => 'that contains the files that you want copy randomized to the usb stick',
	'that means' => 'that means',
	'this work is licensed under a' => 'this work is licensed under a',
	'to' => 'to',
	'was copied' => 'was copied',
	'work_in_progress' => 'work in progress'
);

$lang['ger'] = array(
	'after start' => 'Nach dem Starten ist es nicht nötig auf eine Antwort des Tools zu warten. Du kannst den Browser einfach schließen. Du bekommst nach dem Kopieren eine Meldung per E-Mail',
	'or' => 'oder',
	'copy in progress' => 'Beim kopieren',
	'please check progressinformation on startpage' => 'Der Fortschritt kann über die Startseite eingesehen werden',
	'startcopyprocess' => 'Starte Kopiervorgang',
	'back to startpage' => 'Zurück zur Hauptseite',
	'added to job' => 'Zum kopierauftrag hinzugefügt',
	'all' => 'alles',
	'any issue' => 'irgendeine st&ouml;rung',
	'are available' => 'sind verf&uuml;gbar',
	'are to big for destination' => 'zu gro&szlig; f&uuml;r das ziel',
	'audioformats' => 'audio formate',
	'autor' => 'author',
	'breakes the copyjob at first to big file' => 'bricht den kopiervorgang bei der ersten zu gro&szlig;en datei ab',
	'but only' => 'aber nur',
	'but' => 'aber',
	'clean' => 'bereinigen',
	'contains subfolders' => 'unterverzeichnisse enth&auml;lt',
	'convert files to mp3' => 'konvertiere dateien nach mp3',
	'copy 50 files in each folder cd01 - cd06 to use it with a cd changer simulator for older carradios. as example when you use the yatour digital music changer.' => 'kopiere 50 musikdateien in die jeweiligen verzeichnisse cd01 - cd06 zur nutzung in einem cd wechsel simulator für &auml;ltere autoradios. z.b. wenn f&uuml;r die nutzung mit dem yatour digital music changer',
	'copy soundfiles to usb stick' => 'kopiert musikdateien auf einen usb stick',
	'could not copy' => 'konnte nicht kopiert werden',
	'count' => 'anzahl',
	'delete content' => 'l&ouml;sche inhalt',
	'destination file are not writable' => 'zieldatei ist nicht schreibbar',
	'destination folder not exist' => 'zielverzeichnis exitsiert nicht',
	'destination is not allowed, maybe you forgot to choose a destination or the stick is not recognize from ds' => 'ziel ist nicht erlaubt, ggf wurde vergessen das ziel anzugeben oder es wurde keines von der ds erkannt',
	'destionations' => 'ziel',
	'done' => 'kopieren abgeschlossen',
	'feedback' => '<a href="mailto:thiemo@schuff.eu&subject=loadmusicstick&body=Hallo Thiemo,...">E-Mail</a> oder <a href="http://www.synology-forum.de/showthread.html?58610-neue-App-LoadMusicStick" target="_blank">Forum</a>',
	'file size is' => 'dateigr&ouml;&szlig;e betr&auml;gt',
	'file' => 'datei',
	'folder' => 'verzeichnis',
	'free space on destination' => 'freier speichplatz auf dem ziel',
	'genres' => 'genres',
	'in reason of' => 'aufgrund von',
	'left free space on destination' => 'auf ziellaufwerk frei lassen',
	'lets go' => 'los geht\'s',
	'license' => 'lizenz',
	'no folders with sources files choosen' => 'kein verzeichnis mit quelldateien gew&auml;hlt',
	'no portable drive as destination found' => 'kein Ziellaufwerk erkannt',
	'no source avaiable, please check if folder' => 'keine quelle gefunden, bitte pr&uuml;fe ob',
	'no write permission' => 'keine schreibrechte',
	'none' => 'nichts',
	'open synology filemanager' => '&ouml;ffne synology dateimanager',
	'option' => 'optionen',
	'permission' => 'berechtigung',
	'protocol' => 'protokol',
	'select the destination of your music files' => 'w&auml;hle das ziel deiner musikdateien',
	'select the recognize folders from' => 'w&auml;hle die erkannten verzeichnisse von',
	'select' => 'w&auml;hle',
	'settings for the kind of copy' => 'einstellungen f&uuml;r die art des kopierens',
	'size' => 'gr&ouml;&szlig;e',
	'source file is not a file' => 'quelldatei ist keine datei',
	'source file is not readable' => 'quelldatei ist nicht lesbar',
	'source file now exist' => 'quelldatei existiert nicht',
	'source' => 'quelle',
	'start to copy' => 'starte zum kopieren',
	'that contains the files that you want copy randomized to the usb stick' => 'welche die musikdateien enthalten die du per zufall auf deinen usb stick kopiert haben m&ouml;chtest',
	'that means' => 'das entspricht',
	'this work is licensed under a' => 'dieses werk unterliegt einer',
	'to' => 'nach',
	'was copied' => 'wurde kopiert',
	'differfolderbygenre' => 'erstellt eigenes verzeichnis für jedes Genre',
	'work_in_progress' => 'in Arbeit'
);

?>